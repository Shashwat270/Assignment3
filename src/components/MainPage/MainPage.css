body, html {
    height: 100%;
    width: 100%;
    font-weight: 10;
    margin: 0px;
  }
  .page-theme {
      background-color: #4a4b4c;
  }
  .content-width {
    width: 80%;
    margin: auto;
  }
  .div-books {
    display: flex;
    justify-content: center;
    flex-wrap: wrap; 
    align-content: space-between;
    flex: 1 0 calc(33.333% - 20px);
  }