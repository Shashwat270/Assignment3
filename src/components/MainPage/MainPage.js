import React, { Component } from 'react';
import './MainPage.css';
import AllBooks from '../BooksOfAAuthor/AllBooks';
import Header from '../Header/Header';
import Authorname from '../AuthorName/AuthorName';

import getData from '../../UtilityFunction/getData';

class MainPage extends Component {
    state = {
      bookData: {},
    }

    componentDidMount() {
      getData('http://localhost:8080/books').then(response => this.storeData(response));
    }

    storeData = responseObject => this.setState({ bookData: responseObject });

    render() {
      return (
        <div className="page-theme">
          <Header />
          <div className="content-width div-books" />
          <AllBooks books={this.state.bookData['J K Rowling']} />
          {/* <AllBooks books={booksByWriter[1]} /> */}
        </div>
      );
    }
}
export default MainPage;
