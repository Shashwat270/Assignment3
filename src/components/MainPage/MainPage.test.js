import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

describe('CreateBlogForm', () => {
  it('should render form for creating Blog Post', () => {
    const tree = renderer.create(<MainPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
