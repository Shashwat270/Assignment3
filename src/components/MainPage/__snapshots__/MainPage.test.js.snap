// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBlogForm should render form for creating Blog Post 1`] = `
<div
  className="page-theme"
>
  <ul
    className="header"
  >
    <li>
      <h1>
        BS
      </h1>
    </li>
  </ul>
  <div
    className="content-width div-books"
  >
    <div
      className="card"
    >
      <img
        alt="Book"
        className="book-image"
        src="cover_image.jpeg"
      />
      <p
        className="book-name"
      >
        Harry Potter and the sorcerer stone
      </p>
      <span
        className="rating-data"
      >
        4.5
      </span>
      <img
        alt="Heart"
        className="heart-logo"
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "heart-red.svg",
          }
        }
      />
    </div>
  </div>
</div>
`;
